aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu

```diff
@@ -49,68 +51,75 @@ namespace {
   }
 }
 
+void s_addmm_out_csr_sparse_dense_cuda_worker(int64_t nnz, int64_t m, int64_t n, int64_t k, Tensor& r_, const Scalar& beta, const Tensor& t, const Scalar& alpha, Tensor& crow_indices, Tensor& col_indices, Tensor& values, const Tensor& dense) {
+
+  // No half support, so we don't have to use CUDATypeConversion
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(
+      values.scalar_type(), "addmm_sparse_cuda", [&] {
+        scalar_t cast_beta = beta.to<scalar_t>();
+        scalar_t cast_alpha = alpha.to<scalar_t>();
+        Tensor r__;
+        if (cast_beta == scalar_t(0)) {
+          r_.zero_();
+        } else if (!is_same_tensor(t, r_)) {
+          r_.copy_(t);
+        }
+        if (nnz > 0) {
```

Comment:
With `if (nnz == 0)` used in the caller functions, this `if` condition can now be eliminated.
Use `TORCH_INTERNAL_ASSERT(nnz > 0)` instead.